lib/settings.js

```diff
@@ -94,12 +94,30 @@ class Settings extends EventEmitter {
       this._defaultSettingsFilePath;
   }
 
+  _buildWatchPaths(path) {
+    const parts = path.split('.');
+    const paths = [];
+
+    for (let i = parts.length; i > 0; i--) {
+      const path = _.take(parts, i);
+      paths.push(_.join(path, '.'));
+    }
+
+    return paths;
+  }
+
   _handleObjectChanged(keyPath = '') {
     const settingsFilePath = this._getSettingsFilePath();
     jsonfile.writeFileSync(settingsFilePath, this._cacheObj);
-    const relatedHandler = keyPath ?
-      _.filter(this._watchHandlers, f => f.keyPath.startsWith(keyPath)) :
-      this._watchHandlers;
+    let relatedHandler = [];
+    if (keyPath) {
+      const paths = this._buildWatchPaths(keyPath);
+      relatedHandler =
+      _.filter(this._watchHandlers, f => _.some(paths, c => f.keyPath.startsWith(c)));
```

Comment:
I don't get why we need to do the path splitting, can you give a specific example of how your change catches and the old one doesn't?

The test case that you have seems to suggest that it'll still work for old code? (ex: without splitting)

Reply:
Okay, that's not what's included in the testing though.

Also, if this is what we are testing, shouldn't the following work as well?
```
actual path.startWith(watch key) || watch key.startWith(actual path)
```

Reply:
ah...i think it will work as well.